src/Microsoft.AspNetCore.Mvc.Formatters.Json/JsonInputFormatter.cs

```diff
@@ -158,7 +158,18 @@ public class JsonInputFormatter : TextInputFormatter
 
                     if (successful)
                     {
-                        return InputFormatterResult.SuccessAsync(model);
+                        if (model == null && !context.AllowEmptyInput)
```

Comment:
> In what circumstance would someone be surprised by this behavior?

We've been saying since the start of this issue that *"`[FromBody]` should imply 'required'"*. If we teach people that, and they also believe that "*required means I get non-null or `IsValid`==`false`*", then they may well write code that assumes their `[FromBody]` param isn't going to receive null when `IsValid`==`true`. They might even test it with a blank request and see the framework appears to be protecting them, and be happy. But they will not be happy if they later find that a crafty user sends either whitespace or the literal text "`null`" and manages to get a `null` into the method while also `IsValid`==`true`.

Are there other common cases where a property marked as required for model binding might arrive as null while also `IsValid`==`true`? If so, maybe this isn't a big deal. But if this is the *only* way that happens, it would be pretty nasty.